src/components/tables/RowActions/Shared/UpdateEntity.tsx

```diff
@@ -43,10 +43,7 @@ function UpdateEntity({
     const [pubID, setPubID] = useState<string | null>(null);
     const [logToken, setLogToken] = useState<string | null>(null);
 
-    const { liveSpecs } = useLiveSpecsExtWithSpec(
-        entity.last_pub_id,
-        entity.spec_type
-    );
+    const { liveSpecs } = useLiveSpecsExtWithSpec(entity.id, entity.spec_type);
```

Comment:
Querying by the `last_pub_id` only works when users are just using the UI and only publishing things individually. This breaks right away if someone uses `flowctl` to publish multiple items at once. This _can_ cause the UI to fetch the wrong spec when trying to update the `shards` settings.

We should query by the `id` so we know we are fetching the correct spec